storage/Devicegraph.cc

```diff
@@ -43,13 +44,15 @@ namespace storage
 
 
     WrongNumberOfParents::WrongNumberOfParents(size_t seen, size_t expected)
-	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected))
+	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected),
+		    Silencer::is_any_active() ? DEBUG : WARNING)
```

Comment:
The purpose of that Silencer class is to prevent Storage::Exception subclasses from logging when I know that there might be exceptions, but I am watching out carefully for them, catch them and then act accordingly.

The general idea is to be able to react to special situations without flooding the log with warnings that there were exceptions. The calling code has to take care about it. This is the contract: I take responsibility for what's going to happen and you just shut up for a while - until I tell you otherwise.

silencer = ::Storage::Silencer.do
  begin
    partitions = disk.partition_table.partitions
  rescue ::Storage::Exception => ex
    log.info("No partition table on #{disk}")
  end
end

For that do..end block, Storage exceptions will be silent. They will not flood the log. That's all that class does. And that's the underlying contract.